### PR TITLE
DATAUP-314: misc decruftification

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Some bits of cruft that I found whilst implementing the batch job modals. There is a lot of duplication in the files under `nbextensions`, and I compared those with the same name to see if they were identical or not. At the same time, I fixed some eslint errors and altered formatting in places to make them as similar as possible to aid in my duplicate-finding quest.

The result:
- duplicated modules: I moved one copy into `js/common/cellComponents/` and updated the references elsewhere
- reams of commented-out code: deleted
- minor eslint fixes

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-314
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and playing around, seeing that nothing has changed

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: nbextensions/advancedViewCell/widgets/appParamsViewWidget.js [hunk 1]
```diff
@@ -503,147 +460,129 @@ define([
...
-                                                model.getItem(['params', spec.id])
-                                            ).then((widget) => {
-                                                widgets.push(widget);
-
-                                                return widget.start({
-                                                    node: document.getElementById(
-                                                        inputParams.view[parameterId].id
-                                                    ),
-                                                });
+                return Promise.resolve()
```
Comment (ialarmedalien) [inline nbextensions/advancedViewCell/widgets/appParamsViewWidget.js:463]: remove unnecessary bracket, which decreases the indentation of this whole section by one tab,

File: nbextensions/advancedViewCell/widgets/appParamsViewWidget.js [hunk 2]
```diff
@@ -409,25 +385,6 @@ define([
...
-            // and outputs with output
-
-            // and params with param
-
-            // validate type
-
-            return spec;
-        }
-
-        function validateParameterSpecs(params) {
```
Comment (ialarmedalien) [inline nbextensions/advancedViewCell/widgets/appParamsViewWidget.js:425]: unused => deleted

File: kbase-extension/static/kbase/js/common/cellComponents/appInfoDialog.js [hunk 3]
```diff
@@ -44,21 +44,24 @@ define([
...
-                        data.authors.length > 1 ? 's' : '',
+                        data.authors.length === 1 ? '' : 's',
                         ': ',
-                        span(
-                            { class: 'value' },
-                            (function () {
-                                if (data.authors.length === 1) {
-                                    return data.authors[0];
-                                }
-                                return ul(data.authors.map(li));
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/cellComponents/appInfoDialog.js:55]: this was just creating a list of numbers with no names!

File: kbase-extension/static/kbase/js/common/cellComponents/appInfoDialog.js [hunk 4]
```diff
@@ -116,10 +119,8 @@ define([
                 });
             })
             .then((result) => {
-                switch (result.action) {
-                    case 'link':
-                        window.open(result.result.url, result.result.name);
-                        break;
+                if (result.action === 'link') {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/cellComponents/appInfoDialog.js:122]: only one option -- no need to use `switch`

File: kbase-extension/static/kbase/js/common/cellComponents/inputWrapperWidget.js [hunk 5]
```diff
@@ -1,10 +1,10 @@
-define(['kb_common/html'], (html) => {
+define(['common/html'], (html) => {
     'use strict';
     const t = html.tag,
         div = t('div');
     function factory(config) {
-        let container,
-            wrappedWidget = config.widget,
+        let container;
+        const wrappedWidget = config.widget,
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/cellComponents/inputWrapperWidget.js:7]: eslint

File: nbextensions/advancedViewCell/widgets/appParamsViewWidget.js [hunk 6]
```diff
@@ -140,7 +136,7 @@ define([
                     key: {
                         type: 'get-param-state',
                     },
-                    handle: function (message) {
```
Comment (ialarmedalien) [inline nbextensions/advancedViewCell/widgets/appParamsViewWidget.js:143]: `message` is not used

File: nbextensions/advancedViewCell/widgets/appParamsViewWidget.js [hunk 7]
```diff
@@ -240,24 +232,19 @@ define([
...
             let message;
+            let showAdvancedButton;
             if (settings.showAdvanced) {
                 if (advancedInputs.length > 1) {
                     message = String(advancedInputs.length) + ' advanced parameters showing';
                 } else {
                     message = String(advancedInputs.length) + ' advanced parameter showing';
                 }
-                var showAdvancedButton = ui.buildButton({
+                showAdvancedButton = ui.buildButton({
```
Comment (ialarmedalien) [inline nbextensions/advancedViewCell/widgets/appParamsViewWidget.js:247]: remove `var` as `showAdvancedButton` is initialised above

**Review by briehl — APPROVED** [review]
Yup, pretty straightforward. 👍

**Pull request merged**